src/components/events/eventFeed/eventFeed.tsx

```diff
@@ -18,33 +18,29 @@ import {
 import { userRole } from '../../../types/string-types';
 
 interface eventProps {
-    eventId: string | undefined;
     userRole: userRole;
+    handleAlert: (isSuccess: boolean, content: string) => void;
 }
 
-const Event = ({ eventId, userRole }: eventProps) => {
+const Event = ({ userRole, handleAlert }: eventProps) => {
   const [event, setEvent] = useState<IEvent>();
   const [loading, setLoading] = useState<boolean>(false);
   const [isEdit, setIsEdit] = useState<boolean>(false);
-
-  if (!eventId) {
-    const navigate = useNavigate();
-    navigate('/events');
-    return null;
-  }
+  const { id } = useParams();
 
   const fetchEvent = useCallback(async (): Promise<void> => {
```

Comment:
Im not entirely sure but I think no callback is needed to API req , just use it inside of the useeffect with  IIFE
https://developer.mozilla.org/en-US/docs/Glossary/IIFE

Reply:
Your example does not work because useEffect cannot be an async function.
I fixed it in another way.